### PR TITLE
[chore] [receiver/datadog] Add support for v2 series

### PR DESCRIPTION
**Pull request by carrieedwards**
**Description:**
This PR adds support for Datadog V2 series.

Follow up of #33631 and #33957.

The full version of the code can be found in the `cedwards/datadog-metrics-receiver-full` branch, or in Grafana Alloy: https://github.com/grafana/alloy/tree/main/internal/etc/datadogreceiver

**Link to tracking Issue:** 
#18278 

**Testing:** 
Unit tests, as well as an end-to-end test, have been added.

### CODE REVIEW

**Review by jesusvazquez — APPROVED** [review]
LGTM!

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.21.0
 
-toolchain go1.21.12
+toolchain go1.22.5
```
Comment (jpkrohling) [inline cmd/otelcontribcol/go.mod:7]: This is my first PR review after the (long) PTO, but perhaps this change shouldn't be here?

File: cmd/oteltestbedcol/go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/oteltestbed
 
 go 1.21.0
 
-toolchain go1.21.12
+toolchain go1.22.5
```
Comment (jpkrohling) [inline cmd/oteltestbedcol/go.mod:7]: Same here.

Comment (jpkrohling) [inline cmd/oteltestbedcol/go.mod:7]: Same here.

File: receiver/datadogreceiver/internal/translator/series.go [hunk 3]
```diff
@@ -87,3 +107,68 @@ func (mt *MetricsTranslator) TranslateSeriesV1(series SeriesList) pmetric.Metric
...
+
+	for _, serie := range series {
+		var dps pmetric.NumberDataPointSlice
+
+		// The V2 payload stores the host name under in the Resources field
+		resourceMap := getV2Resources(serie.Resources)
+		// TODO(jesus.vazquez) (Do this with string interning)
+		dimensions := parseSeriesProperties(serie.Metric, strings.ToLower(serie.Type.String()), serie.Tags, resourceMap["host"], mt.buildInfo.Version, mt.stringPool)
+		for k, v := range resourceMap {
+			if k == "host" {
```
Comment (jpkrohling) [inline receiver/datadogreceiver/internal/translator/series.go:122]: this likely deserves a code comment -- why is host being specifically skipped?

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**